uvtools/dspec.py

```diff
@@ -60,11 +60,356 @@ def calc_width(filter_size, real_delta, nsamples):
         lthresh = nsamples
     return (uthresh, lthresh)
 
-
+def fourier_filter(x, data, wgts, filter_centers, filter_half_widths, suppression_factors,
+                   mode, filter2d, fitting_options, cache=None, filter_dim=1, skip_wgt=0.1,
+                   max_contiguous_edge_flags=10):
+                   '''
+                   Your one-stop-shop for fourier filtering.
+                   We don't use the other filtering functions anymore.
+                   It can filter 1d or 2d data with x-axis(es) x and wgts in fourier domain
+                   rectangular windows centered at filter_centers or filter_half_widths
+                   perform filtering along any of 2 dimensions in 2d or 1d!
+                   the 'dft' and 'dayenu' modes support irregularly sampled data.
+                   -----------
+                   x: array-like
+                      Array of floats giving x-values of data. Depending on the chosen method, this data may need to be equally spaced.
+                      If performing a 2d clean, a 2-list or 2-tuple of np.ndarrays with x-values should be provided.
+                    data: array-like
+                        1d or 2d numpy.ndarray of complex data to filter.
+                    wgts: array-like
+                        1d or 2d numpy.ndarray of real weights. Must be the same shape as data.
+                    filter_centers: array-like
```

Comment:
if mode is `clean` then does this need to be passed? perhaps this should be a kwarg? same for other dayenu arg inputs below

Reply:
I'm a little bit confused by your comment here. `clean` still uses `filter_centers` and `filter_half_widths` etc... to determine the areas to be cleaned in the delay transform so I do think these arguments are necessary for `clean` mode. I'm wondering if this comment is referring to something else though.